### PR TITLE
faster consensus catchup

### PR DESCRIPTION
**Pull request by larskuhtz**
* [x] Do catch up in steps of 1000 blocks
* [x] Include origin header into pulled cut hashes

### CODE REVIEW

File: src/Chainweb/CutDB.hs [hunk 1]
```diff
@@ -311,11 +312,24 @@ processCuts logFun headerStore payloadStore queue cutVar = queueToStream
     threshold :: Int
     threshold = int $ 2 * diameter graph * order graph
 
+    -- NOTE: THIS NUMBER MUST BE LARGER THAN THE RESPECTIVE LIMIT
+    -- IN 'CutDB.Sync'
+    farAHeadThreshold :: Int
+    farAHeadThreshold = 2000
```
Comment (fosskers) [inline src/Chainweb/CutDB.hs]: This makes me nervous.

File: src/Chainweb/CutDB/Sync.hs [hunk 2]
```diff
@@ -95,9 +99,12 @@ syncSession v useOrigin p db logg env = do
...
 
     receive = do
-        c <- getCut cenv
-        logg @T.Text Debug $ "got cut " <> sshow c
-        addCutHashes db c
+        -- Query cut that is at most 1000 blocks ahead
+        h <- _cutHeight <$> _cut db
+        c <- getCut cenv (h + 1000)
+
+        logg @T.Text Info $ "receivecd cut " <> sshow c
```
Comment (fosskers) [inline src/Chainweb/CutDB/Sync.hs]:
```suggestion
        logg @T.Text Info $ "received cut " <> sshow c
```

File: src/P2P/TaskQueue.hs [hunk 3]
```diff
@@ -140,12 +143,22 @@ instance Ord (Task env a) where
 -- enough to be useful in other contexts, too.
 --
 session
+    :: AttemptsCount
+    -> PQueue (Task env a)
+    -> LogFunction
+    -> env
+    -> PeerInfo
+    -> IO Bool
+session li q lo e _ = session_ li q lo e
```
Comment (fosskers) [inline src/P2P/TaskQueue.hs]: Why not consolidate on a single `session` function, if it's just redirecting like this anyway?

Comment (fosskers) [reply src/P2P/TaskQueue.hs]: Ah, it's that the `P2pSession` type changed. Hm.

File: src/Chainweb/Utils/Paging.hs [hunk 4]
```diff
@@ -129,6 +132,15 @@ data NextItem k
     | Exclusive k
     deriving stock (Eq, Show, Ord, Functor, Foldable, Traversable)
 
+_getNextItem :: NextItem k -> k
+_getNextItem (Inclusive k) = k
+_getNextItem (Exclusive k) = k
+{-# INLINE _getNextItem #-}
+
+getNextItem :: Getter (NextItem k) k
+getNextItem = to _getNextItem
```
Comment (fosskers) [inline src/Chainweb/Utils/Paging.hs]: This throws a redundant constraint warning.

File: src/Chainweb/CutDB.hs (hunk 1 above)
Comment (larskuhtz) [reply src/Chainweb/CutDB.hs]: This will be fixed in the Rocksdb PR #163

File: src/P2P/TaskQueue.hs (hunk 3 above)
Comment (larskuhtz) [reply src/P2P/TaskQueue.hs]: The `session_` function is there mostly to limit the amount of code changes and not have to wrap `PeerInfo` in a `Maybe` for situations when it's not available.

File: src/Chainweb/Utils/Paging.hs (hunk 4 above)
Comment (larskuhtz) [reply src/Chainweb/Utils/Paging.hs]: Yes. Didn't the team agree sometime ago to remove the redundant-constraint warning flag from the cabal file?

Comment (larskuhtz) [reply src/Chainweb/Utils/Paging.hs]: I guess removing it from the cabal file should be in a separate PR. So I'll add the the pragma to suppress it.

Comment (larskuhtz) [reply src/Chainweb/Utils/Paging.hs]: accidentally committed this into #163. So, won't fix it in the PR.

**Pull request merged**